### PR TITLE
Avoid env reload

### PR DESCRIPTION
**Pull request by o-**

### CODE REVIEW

**Comment by o-** [comment]
@JanJecmen this PR is only for the last commit

File: rir/src/compiler/analysis/last_env.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+namespace pir {
+
+class LastEnv : public StaticAnalysis<AbstractUnique<Value>> {
+  public:
+    LastEnv(ClosureVersion* cls, Code* code, LogStream& log)
+        : StaticAnalysis("Last Env", cls, code, log) {}
+
+    AbstractResult apply(AbstractUnique<Value>& state,
+                         Instruction* i) const override {
+        if (i->hasEnv() && !MkEnv::Cast(i) && i->env() != state.get()) {
```
Comment (JanJecmen) [inline rir/src/compiler/analysis/last_env.h:17]: I am still confused by `MkEnv` vs. `LdFunctionEnv`... This looks like exactly the same condition as the p2r compiler `explicitEnvValue`. Why is one explicit and the other not?
Or, shouldn't this test also check the other one?

**Review by JanJecmen — APPROVED** [review]
lgtm!

File: rir/src/compiler/analysis/last_env.h (hunk 1 above)
Comment (o-) [reply rir/src/compiler/analysis/last_env.h:17]: This is about the arguments. For almost all instruction an environment argument means, that the env should be set using `set_env_`. Except `mk_env_`, this one expects the parent env as a normal argument on the stack and not as the value of `*env`.

**Pull request merged**